### PR TITLE
Adding ability to set Service Links in Pod Spec

### PR DESCRIPTION
**Pull request by amielkelinkedin**
Adding Ability to set Service Links In Pod Spec Template allowing for the modification of Kubernetes Behavior, especially ENV variablles.

### CODE REVIEW

**Comment by amielkelinkedin** [comment]
This builds locally, and runs on internal servers.

**Review by inramana — DISMISSED** [review]
LGTM

File: az-core/src/main/java/azkaban/Constants.java [hunk 1]
```diff
@@ -692,6 +692,9 @@ public static class ContainerizedDispatchManagerProperties {
     public static final String KUBERNETES_POD_SERVICE_ACCOUNT_TOKEN_AUTOMOUNT =
         KUBERNETES_POD_PREFIX + "service.account.token.automount";
 
+    public static final String KUBERNETES_POD_SERVICE_LINKS =
+        KUBERNETES_POD_PREFIX + "service.account.links";
```
Comment (sshardool) [inline az-core/src/main/java/azkaban/Constants.java:696]: Consider renaming to something like `inject.service.links` ?
This is not service account related and could be confusing.

**Review by ypadron-in — APPROVED** [review]
LGTM

**Review by sshardool — COMMENTED** [review]
Looks good overall, with one comment. 
Also  - please add a unit test for this (can be in a follow up commit if this is urgent)

File: az-core/src/main/java/azkaban/Constants.java (hunk 1 above)
Comment (amielkelinkedin) [reply az-core/src/main/java/azkaban/Constants.java:696]: Will just change to service links. thanks!

**Review by sshardool — APPROVED** [review]

**Pull request merged**